src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c

```diff
@@ -738,154 +758,414 @@ int xocl_cleanup_mem(struct xocl_drm *drm_p)
 	return ret;
 }
 
-int xocl_init_mem(struct xocl_drm *drm_p)
+static int init_mem_group(struct connectivity *connect, int ip_cnt, int arg_idx, 
```

Comment:
could we add some sanity check against the connectivity info to make sure an argument can only connect to a range of memory banks and there is no holes in the range. say, you should not connect to bank 1 and 3 without connecting to bank 2. we need to fail fast and loudly if we see this. i don't trust xclbin meta data that much...

Reply:
Added sanity check as suggested.